backend/services/goals_service.py

```
@@ -231,3 +238,116 @@ def check_and_populate_goals(self, user_id: int):
+            user_id=user_id,
+            goal_id=goal.id,
+            date_assigned=date_assigned,
+        )
+        db.session.add(user_goal)
+        db.session.commit()
+
+        logger.info("Personalized goal added successfully")
+        return {"message": "Personalized goal added successfully"}, 200
+
+    @staticmethod
+    def _create_goal_title(type, requirement):
```

Ruff (0.8.2)

Missing return type annotation for staticmethod `_create_goal_title`

Add return type annotation: `Optional[str]`

(ANN205)